src/pivot-table/data/extract-headers.ts

```diff
@@ -44,6 +44,7 @@ const extractHeaders = (
         qApprMaxGlyphCount: qDimensionInfo.qApprMaxGlyphCount,
         isDim: true,
         headTextAlign: "left",
+        isAncestorPseudoDimension: matrix[rowCount - 1].some((cell) => cell && !cell.isDim),
```

Comment:
This will be true no matter where in the "hierarchy" the pseudo dimension is position. So not really `isAncestorPseudoDimension` in that sense. But maybe that is the logic that is needed?

Reply:
Also consider declaring this so that it can be reused `const row = matrix[rowCount - 1];`

Reply:
I don't think so. I loop over the entire row, since we first create the matrix containg null. But when the you create cell n, every cell > n is null. So if the pseudo dim hasn't been added yet, `isAncestorPseudoDimension` should be false

Reply:
Or do you mean that it isn't really an "ancestor". Yeah that is true, since it is not a tree. I went with the same naming for consistency. I think it would make sense that `HeaderCell` and `Cell` have more common properties. using `x` and `y` instead of `colIdx` and `rowIdx` for example. But I know HeaderCell is aligned with the menu props. We would have to take a more thurough look at that if we want to rename things again

Reply:
Ah yes, nice 👍 

I do believe the time complexity is O(n^2). Where the worse case would be lots of dimensions with no pseudo dim. I doubt that is a problem but if so, you could probably store the index of the pseudo dim and compare that with the `colIdx` instead. That should then be O(n).

Reply:
Btw I think @a-m-dev could use `isAncestorPseudoDimension` in #437 to simplify the qDimensionInfo index calculation?

Reply:
kind of, basically i just need an (actual) index of the dim which engine would interpret it correctly. 
So to be exact maybe not this variable of `isAncestorPseudoDimension` but the thing you mentioned that you would add to cell and I could then use to get the correct index without my calculations should be fine to make sort work correctly!